modules/light-clients/09-localhost/light_client_module.go

```diff
@@ -144,6 +144,21 @@ func (LightClientModule) Status(ctx sdk.Context, clientID string) exported.Statu
 	return exported.Active
 }
 
+// LatestHeight returns the latest height for the client state for the given client identifier.
+// If no client is present for the provided client identifier a zero value height is returned.
+//
+// CONTRACT: clientID is validated in 02-client router, thus clientID is assumed here to be 09-localhost.
+func (lcm LightClientModule) LatestHeight(ctx sdk.Context, clientID string) exported.Height {
+	clientStore := lcm.storeProvider.ClientStore(ctx, clientID)
+
+	clientState, found := getClientState(clientStore, lcm.cdc)
+	if !found {
+		return clienttypes.ZeroHeight()
+	}
+
+	return clientState.LatestHeight
```

Comment:
This should just return the context block height, but we can readjust later I suppose? Might be better that way as we could add a migration to cleanup state. With this change, we can stop storing state for the localhost and stop doing disc reads/writes for every block for it

Reply:
Definitely, would support leaving this as is for now and addressing localhost in isolation! 👍🏻

Reply:
#5959